torch/utils/cpp_extension.py

```diff
@@ -148,6 +148,7 @@ class BuildExtension(build_ext):
     def build_extensions(self):
         self._check_abi()
         for extension in self.extensions:
+            self._add_compile_flag(extension, '-DTORCH_WITH_PYTHON_BINDINGS')
```

Comment:
I'm not a big fan of macros in this style, because they are "infectious": if I write a header that includes `torch.h` and I build my code with `-DTORCH_WITH_PYTHON_BINDINGS`, *everyone* downstream must also specify this macro.

I agree that having two header files named `torch/torch.h` is suboptimal, and agree it should be fixed. What I don't understand is why `torch/torch.h` has to either be Python-enabled or not, depending on context. Wouldn't it make more sense for there to be a Torch-with-Python header?